source/agora/node/BlockStorage.d

```diff
@@ -1053,24 +1054,13 @@ public class MemBlockStorage : IBlockStorage
         if (finds.empty)
             return false;
 
-        block = deserializeFull!Block(this.blocks[finds.front.position]);
+        try block = deserializeFull!Block(this.blocks[finds.front.position]);
+        catch (Exception e) return false;
```

Comment:
Could also do:

```suggestion
        return collectException(deserializeFull!Block(this.blocks[finds.front.position]) is null);
```

Or keep it your way if you think it makes the intention more clear.

Reply:
The thing I don't like about those utilities (`enforce`, `collectException`) is that they generally don't play well with the compiler.
E.g. the following won't compile:
```D
string foo (int v)
{
    if (v == 42)
        return "42";
    enforce(false, "Expected the answer to life, universe, and the rest");
}
```

However this will:
```D
string foo (int v)
{
    if (v == 42)
        return "42";
    throw new Exception("Expected the answer to life, universe, and the rest");
}
```

Likewise, `collectException` escaping the reference to the `Exception` means it won't play well with DIP1008.